server/metric.go

```diff
@@ -263,6 +263,37 @@ func StatsGetProxiesByType(proxyType string) []*ProxyStats {
 	return res
 }
 
+func StatsGetProxiesByTypeAndName(proxyType string, proxyName string) (res *ProxyStats) {
```

Comment:
It's not efficient to iterate all proxy stats, but ok now. We should optimize this later.